pkg/store/event/event.go

```diff
@@ -428,6 +428,27 @@ func (e *eventStore) DeleteByFinishTime(_ context.Context, ttl time.Duration) er
 	return nil
 }
 
+// DeleteByUID deletes events by the uid of the experiment.
+func (e *eventStore) DeleteByUID(_ context.Context, uid string) error {
+	eventList, err := e.ListByUID(context.Background(), uid)
+	if err != nil {
+		return err
+	}
+	for _, et := range eventList {
+		if err := e.db.Model(core.Event{}).Unscoped().Delete(*et).Error; err != nil {
```

Comment:
I think it's better to batch delete all events after we delete all `PodRecord`.

Reply:
ok. i will modify it.